Client/Frontend/Browser/Playlist/Controllers/PlaylistListViewController.swift

```diff
@@ -664,6 +667,73 @@ extension PlaylistListViewController {
       animated: false,
       scrollPosition: .none)
   }
+  
+  func loadSharedPlaylist(folderSharingUrl: String) {
+    sharedFolderLoadingTask?.cancel()
+    
+    sharedFolderLoadingTask = Task { @MainActor in
+      // Shared Folder already exists
+      if let existingFolder = PlaylistFolder.getSharedFolder(sharedFolderUrl: folderSharingUrl) {
+        PlaylistManager.shared.currentFolder = existingFolder
+        self.loadingState = .fullyLoaded
+        return
+      }
+      
+      // Shared Folder doesn't exist
+      do {
+        self.playerView.setStaticImage(image: UIImage())
+        let model = try await PlaylistSharedFolderNetwork.fetchPlaylist(folderUrl: folderSharingUrl)
+        let folder = await PlaylistSharedFolderNetwork.createInMemoryStorage(for: model)
+        PlaylistManager.shared.currentFolder = folder
+        self.loadingState = .partial
+        
+        if let folderImageUrl = model.folderImage {
+          Task { @MainActor in
+            let authManager = BasicAuthCredentialsManager(for: [folderImageUrl.absoluteString])
+            let session = URLSession(configuration: .ephemeral, delegate: authManager, delegateQueue: .main)
+            defer { session.finishTasksAndInvalidate() }
+            
+            let (data, response) = try await NetworkManager(session: session).dataRequest(with: folderImageUrl)
+            if let response = response as? HTTPURLResponse, response.statusCode != 200 {
+              return
+            }
+            
+            if let image = UIImage(data: data, scale: UIScreen.main.scale) {
+              self.playerView.setStaticImage(image: image)
+            }
+          }
+        }
+        
+        Task { @MainActor in
+          let items = try await PlaylistSharedFolderNetwork.fetchMediaItemInfo(item: model, viewForInvisibleWebView: self.view)
+          try folder.playlistItems?.forEach({ playlistItem in
+            try Task.checkCancellation()
```

Comment:
This cancellation will do nothing because you are creating subtasks and not cancelling those. You need to use `withTaskCancellationHandler` or `async let` to ensure proper cancellation propagates down

Reply:
Addressed. I tested it and it cancels properly.